src/iterative_sorting/iterative_sorting.py

```diff
@@ -1,30 +1,68 @@
-# TO-DO: Complete the selection_sort() function below 
+# TO-DO: Complete the selection_sort() function below
 def selection_sort( arr ):
     # loop through n-1 elements
-    for i in range(0, len(arr) - 1):
+    for i in range(0, len(arr)):
         cur_index = i
         smallest_index = cur_index
-        # TO-DO: find next smallest element
-        # (hint, can do in 3 loc) 
-             
-
-
+        print("smallest_index: ", smallest_index)
+        print("loop: ", cur_index)
+        # Look for smallest value
+        for j in range(i, len(arr)):
+            print("j: ", j)
+            if arr[j] < arr[smallest_index]:
+                # store smallest value index
+                smallest_index = j
+        #temp hold values
```

Comment:
you could get away with having just one temp value
temp = arr[smallest_index]
arr[smallest_index] = arr[cur_index]
arr[cur_index] = temp